src/lm.jl

```diff
@@ -63,18 +63,18 @@ end
 
 cholfact(x::LinearModel) = cholfact(x.pp)
 
-function fit{LmRespT<:LmResp,LinPredT<:LinPred}(::Type{LinearModel{LmRespT,LinPredT}},
-    X::AbstractMatrix, y::Vector)
+function fit{LmRespT<:LmResp,LinPredT<:LinPred, T<:FP}(::Type{LinearModel{LmRespT,LinPredT}},
+    X::@compat(Union{Matrix{T},SparseMatrixCSC{T}}), y::FPVector)
```

Comment:
If `AbstractMatrix` doesn't cause any issues, I'd stick with that here and change `glm` instead (in a separate PR).